### ISSUE TITLE
[BUG] Alphabetical sort is...off

### ISSUE THREAD

**Issue by musictrivianut** [cboxdoerfer/fsearch#250]
**Describe the bug**
Not sure exactly when it started, but fsearch is sorting (name column) by upper case and then lower case.  I believe that it used to be mixed, aA first, bB second.  Now it's [0-9], [A-Z], [a-z], it seems.  Also, I'm trying to remember if the previous sort mixed files and folders or if they were directories first and then files like they are now.  I haven't been using it as regularly as I was a couple of months ago, so hard to tell if that is different.

**To Reproduce**
1. Sort by Name column

**Expected behavior**
Thought that it used to sort differently, [0-9], [aA-zZ].

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04
 - How you installed FSearch: PPA
 - Version: 0.1beta4
 - Output of `localectl status`
   System Locale: LANG=en_US.UTF-8
       VC Keymap: us
      X11 Layout: us,ru
       X11 Model: pc105
     X11 Variant: ,dos
     X11 Options: grp_led:scroll

**Comment by musictrivianut** [comment]
Hmm.  At this point, I'm not sure if it's fsearch or Ubuntu itself.  I just noticed that Nautilus is behaving exactly the same way.  Very weird.

**Comment by cboxdoerfer** [comment]
> I believe that it used to be mixed, aA first, bB second. Now it's [0-9], [A-Z], [a-z], it seems.

I don't think I changed anything in FSearch that would affect that, I've always been using [strverscmp](https://man7.org/linux/man-pages/man3/strverscmp.3.html) to sort filename. On my system this also sorts by `[0-9], [A-Z], [a-z]`, which when I think about it is probably not as useful as `aA - zZ`or `Aa - Zz`.

> Also, I'm trying to remember if the previous sort mixed files and folders or if they were directories first and then files like they are now.

It's always been directories first. Mixing doesn't work currently and it's probably a little tricky to implement, because directories and files are stored separately in the database file and in memory. This means mixing would definitely come with a big performance and maybe memory cost. 

> Hmm. At this point, I'm not sure if it's fsearch or Ubuntu itself. I just noticed that Nautilus is behaving exactly the same way. Very weird.

On my system Nautilus sorts by `aA - zZ` and Dolphin does `Aa - Zz`. I'll give this a little bit more thought and compare it with other implementations before I change anything, but you're right, the current approach is a little weird. Thanks for the report!

**Comment by nyuszika7h** [comment]
Maybe you could make it configurable, but IMO sorting case-insensitively is much more intuitive. I almost never care about whether the first letter is uppercase or lowercase, I just want an alphabetical sort, and sorting uppercase letters first just causes frustration.

**Comment by cboxdoerfer** [comment]
> Maybe you could make it configurable, but IMO sorting case-insensitively is much more intuitive.

Yes, I'm probably going to add a few options for the sort order. Currently it's also not locale and utf8 aware, so in many languages the sort order is just wrong.
